tests/components/template/test_sensor.py

```diff
@@ -438,6 +442,39 @@ def test_available_template_with_entities(self):
         )
 
 
+async def test_creating_sensor_loads_group(hass):
+    """Test setting up template sensor loads group component first."""
+    order = []
+
+    async def async_setup_group(hass, config):
+        # Make sure group takes longer to load, so that it won't
+        # be loaded first by chance
+        await sleep(0.1)
+
+        order.append("group")
+        return True
+
+    async def async_setup_template(
+        hass, config, async_add_entities, discovery_info=None
+    ):
+        order.append("sensor.template")
+        return True
+
+    with patch(
+        "homeassistant.components.group.async_setup", new=async_setup_group,
+    ):
+        with patch(
+            "homeassistant.components.template.sensor.async_setup_platform",
+            new=async_setup_template,
+        ):
+            await async_from_config_dict(
+                {"sensor": {"platform": "template", "sensors": {}}, "group": {}}, hass
+            )
+            await hass.async_block_till_done()
```

Comment:
```suggestion
    with patch(
        "homeassistant.components.group.async_setup", new=async_setup_group,
    ), patch("homeassistant.components.template.sensor.async_setup_platform",
        new=async_setup_template,
    ):
         await async_from_config_dict(
            {"sensor": {"platform": "template", "sensors": {}}, "group": {}}, hass
         )
        await hass.async_block_till_done()
```

..and run it though black

Reply:
Although we really shouldn't be patching home assistant if we can avoid it.

If we can get the `hass.bus.async_listen(EVENT_COMPONENT_LOADED, async_add_to_order)` approach to work it would be more maintainable.